### ISSUE TITLE
Feature request: Amnezia WireGuard configuration support

### ISSUE THREAD

**Issue by marco570** [qdm12/gluetun#2365]
### What's the feature 🧐

Hi! Could you please add Amnezia WireGuard config support. 
Amnezia WireGuard config has additional properties
```
[Interface]
Address = ***
PrivateKey = ***
DNS = ***
MTU = ***
Jc = between 1-128
Jmin = must be less than Jmax
Jmax = less than 1280
S1 = 0
S2 = 0
H1 = 1
H2 = 2
H3 = 3
H4 = 4
```
https://zaneschepke.com/wgtunnel-docs/features.html#amneziawg-support

This feature was added to android wgtunnel app 
https://github.com/zaneschepke/wgtunnel/issues/149

**Comment by marco570** [comment]
Can you add it (in high priority) for ProtonVPN first? 
Because Amnezia WireGuard config compatible with ProtonVPN.

**Comment by codemistake** [comment]
I'm very interested in this functionality, I'll be waiting

**Comment by avt00** [comment]
This is one of the only protocols available in countries with heavy censorship. It's not just for Proton, but for self-hosted solutions too. It would be great if it got higher priorities

**Comment by qdm12** [comment]
@Zhurik did an amazing job with a PR for this https://github.com/qdm12/gluetun/pull/3150 as well as a wiki PR to add this to Gluetun 💯 🎖️ 

You can try it by building it with `docker build -t qmcgaw/gluetun https://github.com/Zhurik/gluetun.git#feature-add-amnezia-wg` and using `WIREGUARD_IMPLEMENTATION=amneziawg` and using this documentation: https://github.com/Zhurik/gluetun-wiki/blob/d7d8e8079e0aa8a524b8af344e072b948186c9af/setup/options/wireguard.md#amnezia-wg

Comments welcome on #3150 whilst the PR isn't merged yet.

**Comment by DX37** [comment]
> [@Zhurik](https://github.com/Zhurik) did an amazing job with a PR for this [#3150](https://github.com/qdm12/gluetun/pull/3150) as well as a wiki PR to add this to Gluetun 💯 🎖️
> 
> You can try it by building it with `docker build -t qmcgaw/gluetun https://github.com/Zhurik/gluetun.git#feature-add-amnezia-wg` and using `WIREGUARD_IMPLEMENTATION=amneziawg` and using this documentation: https://github.com/Zhurik/gluetun-wiki/blob/d7d8e8079e0aa8a524b8af344e072b948186c9af/setup/options/wireguard.md#amnezia-wg
> 
> Comments welcome on [#3150](https://github.com/qdm12/gluetun/pull/3150) whilst the PR isn't merged yet.

Did all that. It's working well for now! HTTP proxy and connected external docker containers.

**Comment by azaglubockii-alt** [comment]
Hi! My issue #3172 was closed as a duplicate of this one.

I just wanted to confirm that with the current latest image (x86_64), AmneziaWG obfuscation is still not effective against my ISP's aggressive DPI (handshake is being dropped with i/o timeout).

I am using the standard J1/J2 and H1/H4 parameters which work fine in the official AmneziaWG desktop app. Is there a specific dev tag or a PR branch I should test to help with the debugging? Thanks!

**Comment by Zhurik** [comment]
@azaglubockii-alt you can build image from [this](https://github.com/qdm12/gluetun/pull/3150) branch:

```shell
docker build -t qmcgaw/gluetun https://github.com/Zhurik/gluetun.git#feature-add-amnezia-wg
```

then change your `AMNEZIA_` parameters to `WIREGUARD_` and `WIREGUARD_IMPLEMENTATION=userspace` to `WIREGUARD_IMPLEMENTATION=amneziawg`

**Comment by azaglubockii-alt** [comment]
Thanks for the update, @Zhurik!

I am using Portainer on my setup, so I cannot easily run docker build locally to test the branch.

Could you please push a pre-built image of this branch to Docker Hub (for example, as qmcgaw/gluetun:amneziawg-test or under your own docker ID)? This would allow me to pull it directly into Portainer and verify the fix with the new WIREGUARD_ environment variables. I'm ready to test it as soon as the image is available!

**Comment by Zhurik** [comment]
@azaglubockii-alt  I built only for linux/amd64

```
zhurik.cr.cloud.ru/gluetun-test:v1
```

if you need for some other platform we can figure something out

**Comment by azaglubockii-alt** [comment]
Hi @Zhurik!

I've updated my stack with your image and the new WIREGUARD_ variables, but I'm getting a pull access denied error in Portainer. It seems the repository zhurik/gluetun is currently set to Private on Docker Hub.

Could you please make it Public so I can pull the amneziawg tag and test it? Thanks!

**Comment by azaglubockii-alt** [comment]
"@Zhurik, I've tried your image zhurik.cr.cloud.ru/gluetun-test:v1 with the new variables. The container starts, but I'm still getting i/o timeout and healthcheck failures. In the logs, it doesn't explicitly say it's using the AmneziaWG engine. Should there be a specific log entry confirming that amneziawg implementation is active?"

**Comment by Zhurik** [comment]
@azaglubockii-alt yeah, it should appear as a `Amneziawg parameters` in Wireguard section of logs

<img width="352" height="255" alt="Image" src="https://github.com/user-attachments/assets/e729bfbd-e01a-423e-8588-a7fc9f8df7cc" />

**Comment by Zhurik** [comment]
Maybe some of your parameters called differently, you can check in [wiki PR](https://github.com/Zhurik/gluetun-wiki/blob/5c7d15d38ed67295b5dbb6613c046348da2e6eac/setup/options/wireguard.md#amnezia-wg)

**Comment by XPhilipp** [comment]
@Zhurik, what a valid config for self hosted amnezia wg?
if:
```
VPN_SERVICE_PROVIDER=custom
VPN_TYPE=amneziawg
AMNEZIAWG_ENDPOINT_IP= ***
AMNEZIAWG_ENDPOINT_PORT= ***
AMNEZIAWG_PUBLIC_KEY=***
AMNEZIAWG_PRIVATE_KEY=***
AMNEZIAWG_ADDRESSES=***
AMNEZIAWG_PRESHARED_KEY=***
AMNEZIAWG_JC = 4
AMNEZIAWG_JMIN = 10
AMNEZIAWG_JMAX = 50
AMNEZIAWG_S1 = 125
AMNEZIAWG_S2 = 124
AMNEZIAWG_S3 = 31
AMNEZIAWG_S4 = 7
AMNEZIAWG_H1 = ***
AMNEZIAWG_H2 = ***
AMNEZIAWG_H3 = ***
AMNEZIAWG_H4 = ***
AMNEZIAWG_I1 = ***
HTTP_CONTROL_SERVER_ADDRESS=:8000
HTTPPROXY=off
SHADOWSOCKS=off
PPROF_ENABLED=no
PPROF_BLOCK_PROFILE_RATE=0
PPROF_MUTEX_PROFILE_RATE=0
PPROF_HTTP_SERVER_ADDRESS=:6060
FIREWALL_ENABLED_DISABLING_IT_SHOOTS_YOU_IN_YOUR_FOOT=on
HEALTH_SERVER_ADDRESS=127.0.0.1:9999
DNS_UPSTREAM_RESOLVERS=cloudflare
LOG_LEVEL=info
STORAGE_FILEPATH=/gluetun/servers.json
PUBLICIP_FILE=/gluetun/ip
VPN_PORT_FORWARDING_STATUS_FILE=/gluetun/forwarded_port
TZ=UTC
```
Error is "ERROR VPN settings: provider settings: server selection: Wireguard server selection settings: endpoint IP is not set"

`WIREGUARD_IMPLEMENTATION=amneziawg`
Does not help too.


P.S. Self build from latest master.


And if adds 
```
WIREGUARD_ENDPOINT_IP=***
WIREGUARD_ENDPOINT_PORT=***
WIREGUARD_PUBLIC_KEY=***
WIREGUARD_PRIVATE_KEY=***
WIREGUARD_ADDRESSES=***
WIREGUARD_PRESHARED_KEY=***
```

<img width="515" height="466" alt="Image" src="https://github.com/user-attachments/assets/491896f8-f2e4-49ef-8b0e-088153ebc780" />

ERROR [vpn] finding a VPN server: VPN type not supported for custom provider: amneziawg

**Comment by Zhurik** [comment]
@XPhilipp you can try fix from this [branch](https://github.com/qdm12/gluetun/pull/3240)

**Comment by XPhilipp** [comment]
> [@XPhilipp](https://github.com/XPhilipp) you can try fix from this [branch](https://github.com/qdm12/gluetun/pull/3240)

Thanks! Will full check tomorrow, but right now:

```
Mar 20 00:07:23 gluetun systemd-networkd[82]: wg0: Link UP
Mar 20 00:07:23 gluetun systemd-networkd[82]: wg0: Gained carrier
```

ip check - well done, from VPS country )

P.S. Cannot found use config file only, but it not necessary for me.
